nevergrad/optimization/differentialevolution.py

```diff
@@ -114,6 +115,13 @@ def __init__(
         self._no_hypervolume = self._config.multiobjective_adaptation
 
     def recommend(self) -> p.Parameter:  # This is NOT the naive version. We deal with noise.
+        sample_size = int((self.dimension * (self.dimension - 1)) / 2 + 2 * self.dimension + 1)
+        if self._config.high_speed and len(self.archive) >= sample_size:
```

Comment:
```suggestion
        if self._config.high_speed and len(self.archive) >= sample_size:
            sample_size = int((self.dimension * (self.dimension - 1)) / 2 + 2 * self.dimension + 1)
```

Reply:
I don't get it --- how can we compare to sample_size before defining sample_size ?

Reply:
my bad, I did not see it was used in the if condition